submissions/Barasii/a-tiny-JS-world-OOP/index.js

```diff
@@ -0,0 +1,86 @@
+class Inhabitant {
+   constructor(species, name, gender, legs, hands, saying, friends) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.hands = hands;
+      this.saying = saying;
+      this.friends = [];
+   }
+
+   toString() {
+      return Object.keys(this).map(prop => `${this[prop]}`).join('; ');
+      }
+
+   callFriends(friends) {         
+      this.friends = friends;
+      }
+}
+
+const Meow = superclass =>
+   class extends superclass {
+      constructor(species, name, gender, legs, hands, saying, friends) {
+         super(species, name, gender, legs, hands, saying, friends);
+         this.saying = 'mur-mur';
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying, friends) {
+      super('human', name, gender, 2, 2, saying, friends);
+   }
+}
+
+class Man extends Human {
+   constructor(name, saying, friends) {
+      super(name, 'male', saying, friends);
+   }
+}
+
+class Woman extends Human {
+   constructor(name, saying, friends) {
+      super(name, 'female', saying, friends);
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor(species, name, gender, saying, friends) {
+      super(species, name, gender, 4, 0, saying, friends);
```

Comment:
If you build a proper hierarchy of classes you won't be needing to pass innatural zero hands.

The idea of classes is to reproduce real life as much as only possible. Animals (beside apes) do not have zero hands, they just do not have any hands, even rudimentary. This is a major architectural flaw that will fire back when you try to scale your world. Make sure you are ready to scaling challenges in your job.